macros/core/PGML.pl

```diff
@@ -1129,6 +1181,46 @@ sub pushItem {
 	}
 }
 
+######################################################################
+
+package PGML::Block::Table;
+our @ISA = ('PGML::Block');
+
+sub pushItem {
+	my $self = shift;
+	my $item;
+	while ($item = shift) {
+		if ($item->{type} eq 'text') {
+			my $text = join('', @{ $item->{stack} });
+			PGML::Warning 'Table text must be in cells' unless $text =~ m/^\s*$/;
+		} elsif ($item->{type} eq 'table-row') {
+			$self->SUPER::pushItem($item);
+		} elsif ($item->{type} eq 'comment') {
+		} else {
+			PGML::Warning 'Tables can contain only table rows';
+		}
+	}
+}
+
+package PGML::Block::TableRow;
+our @ISA = ('PGML::Block');
+
+sub pushItem {
+	my $self = shift;
+	my $item;
+	while ($item = shift) {
+		if ($item->{type} eq 'text') {
+			my $text = join('', @{ $item->{stack} });
+			PGML::Warning 'Table row text must be in cells' unless $text =~ m/^\s*$/;
+		} elsif ($item->{type} eq 'table-cell') {
```

Comment:
It turns out that we need to allow an `options` block to be used on the row stack (it will come after a `table-cell` block`).  This was leading to the options not being ignored (and an error being flagged).  It turns out that PGML doesn't show the error messages, but it probably should.  This should do the trick:

```suggestion
		} elsif ($item->{type} eq 'table-cell' || $item->{type} eq 'options') {
```